File: 704_binary_search.py
class Solution(object):
    def search(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        length = len(nums) - 1
        begin = 0
        end = length
        result = -1
        while begin <= end:
            mid = ((begin + end) / 2)
            if nums[mid] == target:
                return mid
            elif nums[mid] < target:
                begin = mid+1
            else:
                end = mid-1
            
        
        return result

File: 704_binary_search.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int length = nums.size();
        int beginning = 0;
        int end = length-1;
        int mid = 0;
        
        while(beginning <= end){
            mid = beginning + (end - beginning) / 2;
            if(nums[mid] == target){
                return mid;
            }
            else if (nums[mid] < target){
                beginning = mid+1;
            }
            else{
                end = mid-1;
            }
        }
        
        return -1;
    }
};